sherpa/astro/io/tests/test_backend_switch.py

```diff
@@ -0,0 +1,50 @@
+#
+#  Copyright (C) 2021
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+import pytest
+
+from sherpa.utils.testing import requires_data, requires_fits
+from sherpa.astro import io
+from sherpa.astro import ui
+
+
+@requires_fits
+@requires_data
+def test_backend_switch(make_data_path):
+    '''Test that we can switch backends.
+
+    Of course, switching only makes sense if there is more than one backend
+    to try, so this is listed as "requires_fits".
+    '''
+    pha = make_data_path("3c273.pi")
+    ui.load_pha(pha)
+
+    orig_backend = io.backend
+
+    import sherpa.astro.io.dummy_backend
+    io.backend = sherpa.astro.io.dummy_backend
+
+    with pytest.raises(NotImplementedError,
+                       match="No usable I/O backend was imported."):
+        pha = make_data_path("3c273.pi")
```

Comment:
We don't need to re-generate `pha` here (ie this line and line 49) can be removed.

Reply:
Yes, we do. The point here is to test that reading works when a fits backend is present (`@requires_fits`, reading in line 35), then we switch to the dummy backend, so we check in this line that reading now raises an error (that way, we make sure that the backend was actually switched and we don't accidentally continue to use crates or pyfits (which would not issue this error) and then we switch back to the previous backend and check in L 49 that the switch back worked.

We are not reading the pha because we want to do anything, we read it three times to make sure we can or cannot read it with the backend selected at the time.

Reply:
Oh, now I understand; Im just a little dumb this weekend. Please ignore previous comment. Will fix.

Reply:
Oh, now I understand. Please ignore previous comment. Will fix.